src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/ReferralService.kt

```diff
@@ -270,7 +270,7 @@ class ReferralService(
   }
 
   private fun generateReferenceNumber(referral: Referral): String? {
-    val category = referral.intervention.dynamicFrameworkContract.serviceCategory.name
+    val category = referral.intervention.dynamicFrameworkContract.contractType.serviceCategories.elementAt(0).name
```

Comment:
not an issue at all, just out of interest why "elementAt" vs [0]?

Reply:
[0] didn't work on a set.

Reply:
aha!